src/System.Private.Xml/tests/XmlNodeReader/System.Xml.XmlNodeReader.Tests/XmlNodeReaderReadTests.cs

```diff
@@ -0,0 +1,212 @@
+        {
+            string fst = "<!ENTITY fst 'Sample String'>]>";
+            string dtd = "<!DOCTYPE root[<!ELEMENT root (#PCDATA)>" + fst;
+            string xml = dtd + "<root>&fst;</root>";
+            XmlNodeReader nodeReader = NodeReaderTestHelper.CreateNodeReader(xml);
+            Assert.True(nodeReader.Read()); //DTD Read
+            Assert.True(nodeReader.Read()); //Root Element Read
+            Assert.True(nodeReader.Read()); //EntityReference Read
+            Assert.Equal(ReadState.Interactive, nodeReader.ReadState);
+            Assert.Equal(XmlNodeType.EntityReference, nodeReader.NodeType);
+            nodeReader.ResolveEntity();
+            Assert.True(nodeReader.CanResolveEntity);
+            Assert.Equal("Sample String", nodeReader.ReadString());
+        }
+
+        [Fact]
+        public void NodeReaderReadContentAsBase64WithEmptyXml()
+        {
+            var xmlDoc = new XmlDocument();
+            var nodeReader = new XmlNodeReader(xmlDoc);
+            Assert.Equal(0, nodeReader.ReadContentAsBase64(null, 0, 0));
+        }
+
+        [Fact]
+        public void NodeReaderReadContentAsBase64WithSimpleXml()
+        {
+            string xml = "<root attr='val'><child /></root>";
+            XmlNodeReader nodeReader = NodeReaderTestHelper.CreateNodeReader(xml);
+            Assert.True(nodeReader.Read());
+            Assert.True(nodeReader.MoveToAttribute("attr"));
+            Assert.True(nodeReader.CanReadBinaryContent);
+            Assert.Equal(2, nodeReader.ReadContentAsBase64(new byte[33], 10, 10));
+            Assert.Equal(0, nodeReader.ReadContentAsBase64(new byte[33], 10, 10));
+        }
+
+        [Fact]
+        public void NodeReaderReadContentAsBinHexWithEmptyXml()
+        {
+            var xmlDoc = new XmlDocument();
+            var nodeReader = new XmlNodeReader(xmlDoc);
+            Assert.Equal(0, nodeReader.ReadContentAsBinHex(null, 0, 0));
+        }
+
+        [Fact]
+        public void NodeReaderReadContentAsBinHexWithSimpleXml()
+        {
+            string xml = "<root attr='ff0000'><child /></root>";
+            XmlNodeReader nodeReader = NodeReaderTestHelper.CreateNodeReader(xml);
+            Assert.True(nodeReader.Read());
+            Assert.True(nodeReader.MoveToAttribute("attr"));
+            Assert.True(nodeReader.CanReadBinaryContent);
+            Assert.Equal(3, nodeReader.ReadContentAsBinHex(new byte[33], 10, 10));
+            Assert.Equal(0, nodeReader.ReadContentAsBinHex(new byte[33], 10, 10));
+        }
+
+        [Fact]
+        public void NodeReaderReadElementContentAsBase64WithEmptyXml()
+        {
+            var xmlDoc = new XmlDocument();
+            var nodeReader = new XmlNodeReader(xmlDoc);
+            Assert.Equal(0, nodeReader.ReadElementContentAsBase64(null, 0, 0));
+        }
+
+        [Fact]
+        public void NodeReaderReadElementContentAsBase64WithSimpleXml()
+        {
+            string xml = "<root attr='val'>val</root>";
```

Comment:
I'd use some actual base64 value here and make sure conversion happens (also same comment to reading as hex)